include/OpenSpaceToolkit/Astrodynamics/Flight/SatelliteSystem.hpp

```diff
@@ -0,0 +1,106 @@
+
+class SatelliteSystem
+{
+
+    public:
+
+                                SatelliteSystem                             (   const   Mass&                       aMass                                       = { 90.0, Mass::Unit::Kilogram },
+                                                                                const   Matrix3d&                   anInertiaTensor                             = Matrix3d::Identity(),
+                                                                                const   Cuboid&                     aSatelliteGeometry                          = { Point { 0.0, 0.0, 0.0 }, std::array<Vector3d, 3> { Vector3d { 1.0, 0.0, 0.0 }, Vector3d { 0.0, 1.0, 0.0 }, Vector3d { 0.0, 0.0, 1.0 } }, std::array<Real, 3> { 1.0, 2.0, 3.0 } },
+                                                                                const   Real&                       aCrossSectionalSurfaceArea                  = 0.8,
+                                                                                const   Real&                       aDragCoefficient                            = 2.2   ) ;
+
+        SatelliteSystem*        clone                                       ( ) const ;
+
+        bool                    operator ==                                 (   const   SatelliteSystem&            aSatelliteSystem                            ) const ;
+
+        bool                    operator !=                                 (   const   SatelliteSystem&            aSatelliteSystem                            ) const ;
+
+        friend std::ostream&    operator <<                                 (           std::ostream&               anOutputStream,
+                                                                                const   SatelliteSystem&            aSatelliteSystem                                     ) ;
+
+        bool                    isDefined                                   ( ) const ;
+
+        void                    print                                       (           std::ostream&               anOutputStream,
+                                                                                        bool                        displayDecorator                            =   true ) const ;
+
+        Mass                    getMass                                     ( ) const ;
+
+        Matrix3d                getInertiaTensor                            ( ) const ;
+
+        Cuboid                  getGeometry                                 ( ) const ;
+
+        Real                    getCrossSectionalSurfaceArea                ( ) const ;
+
+        Real                    getDragCoefficient                          ( ) const ;
+
+        static SatelliteSystem  DefaultYam                                  ( ) ;
+
+    private:
+
+        Mass                    mass_ ;
+        Matrix3d                inertiaTensor_ ;
+        Cuboid                  satelliteGeometry_ ;
```

Comment:
Yeah you're right that makes sense

Reply:
that makes sense in general, however I implemented crosssectionalarea seperately for now because I didn't want to implement a way of calculating the average crosssectional area from the satellite geometry because its not going to be that simple, and since atmospheric drag is going to be added to ostk soon it makes sense for it to be able to leverage a manually programmed cross sectional area for now. This can be refactored into your idea in the future.